src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -262,6 +262,7 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             if (filterReason == CompletionFilterReason.Insertion &&
                 !IsPotentialFilterCharacter(typeChar) &&
                 !string.IsNullOrEmpty(filterText) &&
+                !string.Equals(filterText, typeChar.ToString(), StringComparison.OrdinalIgnoreCase) &&
                 !Helpers.IsFilterCharacter(bestOrFirstCompletionItem, typeChar, filterText))
```

Comment:
this is a complex enough set of conditions to warrant some explaining comments IMO :)